### PR TITLE
Added new natives:

### PR DESCRIPTION
**Pull request by Donach**
SaveFrameHandle
LoadFrameHandle
BlzGetSpecialEffectScale
BlzSetSpecialEffectMatrixScale
BlzResetSpecialEffectMatrix
BlzSendSyncData
BlzEnableCursor
BlzSetMousePos
BlzGetMouseFocusUnit
BlzGetLocalClientWidth
BlzGetLocalClientHeight
BlzIsLocalClientActive
BlzEnableSelections
BlzIsSelectionEnabled
BlzIsSelectionCircleEnabled
BlzCameraSetupApplyForceDurationSmooth
BlzDisplayChatMessage
BlzGetLocale
BlzEnableTargetIndicator
BlzIsTargetIndicatorEnabled
BlzChangeMinimapTerrainTex

### CODE REVIEW

**Comment by Donach** [comment]
Might change some names of the functions, I tried to match already existing naming, but some of the new functions have weird names...

File: wurst.build [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
```
Comment (Frotty) [inline wurst.build]: don't commit this

File: wurst/_handles/Camera.wurst [hunk 2]
```diff
@@ -81,31 +81,34 @@ public function camerasetup.setDestPosition(vec2 pos, real duration)
 public function camerasetup.getDestPosition() returns vec2
 	return vec2(CameraSetupGetDestPositionX(this),CameraSetupGetDestPositionY(this))
 
-public function camerasetup.apply(boolean doPan, boolean panTimed)
```
Comment (Frotty) [inline wurst/_handles/Camera.wurst]: These boolean->bool changes don't belong in this PR.
Perhaps we should decide a project wide convention, but for now leave them.

File: wurst/_handles/Effect.wurst [hunk 3]
```diff
@@ -206,6 +206,18 @@ public function effect.clearSubAnimations()
 public function effect.removeSubAnimation(subanimtype whichSubAnim)
 	BlzSpecialEffectRemoveSubAnimation(this, whichSubAnim)
 
+/** Get effect scale.*/
+public function effect.getScale() returns real
+	return BlzGetSpecialEffectScale(this)
+
+/** Set effect matrix scale*/
```
Comment (Frotty) [inline wurst/_handles/Effect.wurst]: why is it matrix but takes vector 😕

**Comment by Frotty** [comment]
Nice, looking good 👍

File: wurst/_handles/Effect.wurst (hunk 3 above)
Comment (Donach) [reply wurst/_handles/Effect.wurst]: Is there any three-int tuple? if not, I can create one then...

File: wurst/_handles/Camera.wurst (hunk 2 above)
Comment (Donach) [reply wurst/_handles/Camera.wurst]: will revert it then

File: wurst/_handles/Effect.wurst (hunk 3 above)
Comment (Krypt0n) [reply wurst/_handles/Effect.wurst]: maybe call it public function effect.setScale(vec3 scaleVector)
and forget about blizzard misnaming things
maybe it will be changed after ptr

Comment (Donach) [reply wurst/_handles/Effect.wurst]: Oh, just noticed, nvm my old comment, wil lchange the name a bit

**Comment by Donach** [comment]
Overwritten my previous commit, should fix all things pointed out.

**Comment by Donach** [comment]
This PR is dependent on this https://github.com/wurstscript/WurstScript/pull/848
When that gets pulled, the tests from this PR should start working properly.

**Comment by Donach** [comment]
Since new functions are implemented in WurstScript, could you manually run the build task again?
Thanks!
@Frotty

**Comment by Frotty** [comment]
It runs but your assertions fail:

**Comment by Donach** [comment]
Updated, now it will still fail on assertTrue(g.getUnitAt(g.size()-1) == u2), but that is because there is wrong implementation on the getUnitAt function in Java, not my fault :)
This needs to be fixed first for this to pass the check, but even though, I'd say this can be merged.
@Frotty @Krypt0n - https://github.com/wurstscript/WurstScript/pull/848

**Comment by Frotty** [comment]
Looking good, but I would change all the functions like
```
if a
  return b
return c
```
To ternaries like `a ? b : c`

**Comment by Donach** [comment]
@Frotty Can this be merged?

**Comment by Frotty** [comment]
Yes 😄 🍺
